gcp-observability/src/main/java/io/grpc/gcp/observability/interceptors/LogHelper.java

```diff
@@ -331,67 +318,76 @@ GrpcLogRecord.Builder createTimestamp() {
       Collections.singletonList(
           "grpc-trace-bin"));
 
-  static final class PayloadBuilder<T> {
-    T payload;
-    int size;
+  static final class PayloadBuilderHelper<T> {
+    T payloadBuilder;
     boolean truncated;
 
-    private PayloadBuilder(T payload, int size, boolean truncated) {
-      this.payload = payload;
-      this.size = size;
+    private PayloadBuilderHelper(T payload, boolean truncated) {
+      this.payloadBuilder = payload;
       this.truncated = truncated;
     }
   }
 
-  static PayloadBuilder<GrpcLogRecord.Metadata.Builder> createMetadataProto(Metadata metadata,
+  static PayloadBuilderHelper<Payload.Builder> createMetadataProto(Metadata metadata,
       int maxHeaderBytes) {
     checkNotNull(metadata, "metadata");
     checkArgument(maxHeaderBytes >= 0,
         "maxHeaderBytes must be non negative");
-    GrpcLogRecord.Metadata.Builder metadataBuilder = GrpcLogRecord.Metadata.newBuilder();
-    // This code is tightly coupled with io.grpc.observabilitylog.v1.GrpcLogRecord.Metadata
-    // implementation
-    byte[][] serialized = InternalMetadata.serialize(metadata);
+    Joiner joiner = Joiner.on(", ").skipNulls();
```

Comment:
Why do we need the space after "," ? Was it in the spec? Are all languages doing it?

Reply:
Also did we decide (in the design doc) that multiple values for a single key will be logged as `"key=v1,v2"` instead of `"key=v1"` and `"key=v2"` ?

Reply:
We decided to use a map, so we can't duplicate key names. We should _just_ use a comma; no space.

From https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md
> Duplicate header names may have their values joined with "," as the delimiter and be considered semantically equivalent. Implementations must split Binary-Headers on "," before decoding the Base64-encoded values.

Which is really just re-documenting what is [allowed in HTTP/1](https://www.rfc-editor.org/rfc/rfc7230#section-3.2.2):
> A recipient MAY combine multiple header fields with the same field
   name into one "field-name: field-value" pair, without changing the
   semantics of the message, by appending each subsequent field value to
   the combined field value in order, separated by a comma.

Reply:
@ejona86 what you are describing is behavior on the wire vs how we are logging it. So if RPC1 has `"key=v1,v2"` and RPC2 has `"key=v1"` and `"key=v2"` as 2 separate headers they will get logged exactly the same way with this implementation and that's our conscious decision.

Reply:
Well, the comma-concatenation is visible to applications, because only the application can parse it back into separate headers after the concatenation has been done. RPC1 and RPC2 will be logged identically; the two different encodings are semantically equivalent and the application could have received either form.